src/gasPriceOracle/oracle.ts

```diff
@@ -1,51 +1,93 @@
 import { CHAIN_IDs } from "../constants";
 import { BigNumber, chainIsOPStack } from "../utils";
-import { GasPriceEstimate } from "./types";
+import { GasPriceEstimate, InternalGasPriceEstimate } from "./types";
 import { getPublicClient } from "./util";
 import * as arbitrum from "./adapters/arbitrum";
 import * as ethereum from "./adapters/ethereum";
 import * as linea from "./adapters/linea";
 import * as polygon from "./adapters/polygon";
-import * as arbitrumViem from "./adapters/arbitrum-viem";
 import * as lineaViem from "./adapters/linea-viem";
-import * as polygonViem from "./adapters/polygon-viem";
+
+interface GasPriceEstimateOptions {
+  // baseFeeMultiplier Multiplier applied to base fee for EIP1559 gas prices (or total fee for legacy).
+  baseFeeMultiplier: number;
+  // legacyFallback In the case of an unrecognized chain, fall back to type 0 gas estimation.
+  legacyFallback: boolean;
+  // chainId The chain ID to query for gas prices. If omitted can be inferred by provider.
+  chainId?: number;
+  // unsignedTx The unsigned transaction used for simulation by Viem provider to produce the priority gas fee.
+  unsignedTx?: PopulatedTransaction;
+  // transport Viem Transport object to use for querying gas fees.
+  transport?: Transport;
+}
+
+interface EthersGasPriceEstimateOptions extends GasPriceEstimateOptions {
+  chainId: number;
+}
+
+interface ViemGasPriceEstimateOptions extends Partial<GasPriceEstimateOptions> {
+  baseFeeMultiplier: number;
+}
+
+const GAS_PRICE_ESTIMATE_DEFAULTS: GasPriceEstimateOptions = {
+  baseFeeMultiplier: 1,
+  legacyFallback: true,
+};
 
 /**
  * Provide an estimate for the current gas price for a particular chain.
- * @param chainId The chain ID to query for gas prices.
  * @param provider A valid ethers provider.
- * @param legacyFallback In the case of an unrecognised chain, fall back to type 0 gas estimation.
- * @returns Am object of type GasPriceEstimate.
+ * @param {opts} GasPriceEstimateOptions optional parameters.
+ * @returns An  object of type GasPriceEstimate.
  */
 export async function getGasPriceEstimate(
   provider: providers.Provider,
-  chainId?: number,
-  transport?: Transport,
-  legacyFallback = true
+  opts: Partial<GasPriceEstimateOptions>
 ): Promise<GasPriceEstimate> {
-  chainId ?? ({ chainId } = await provider.getNetwork());
+  const {
+    baseFeeMultiplier,
+    chainId: _chainId,
+    unsignedTx,
+    transport,
+    legacyFallback,
+  }: GasPriceEstimateOptions = {
+    ...GAS_PRICE_ESTIMATE_DEFAULTS,
+    ...opts,
+  };
+  assert(
+    baseFeeMultiplier >= 1.0 && baseFeeMultiplier <= 5,
+    `Require 1.0 < base fee multiplier (${baseFeeMultiplier}) <= 5.0 for a total gas multiplier within [+1.0, +5.0]`
+  );
+
+  const chainId = _chainId ?? (await provider.getNetwork()).chainId;
 
+  // We only use the unsignedTx in the viem flow.
   const useViem = process.env[`NEW_GAS_PRICE_ORACLE_${chainId}`] === "true";
```

Comment:
We only have the linea-viem.ts adapter remaining so we'd better constrain this.

```suggestion
  const useViem = chainId === CHAIN_IDs.LINEA && process.env[`NEW_GAS_PRICE_ORACLE_${chainId}`] === "true";
```